core/src/main/java/org/apache/carbondata/core/index/IndexUtil.java

```diff
@@ -290,6 +291,9 @@ private static FileInputFormat createIndexJob(CarbonTable carbonTable,
       Boolean isFallbackJob, List<String> segmentsToBeRefreshed, boolean isCountJob,
       Configuration configuration) {
     List<String> invalidSegmentNo = new ArrayList<>();
```

Comment:
This is not the only place where we call IndexJobs from, please check it in all the places and add there. Also check it in the case of prepriming too, you can ignore it if it's a non carbon segment.

Reply:
Done. Added in IndexJobs and in pre-priming.